Modules/CPackDeb.cmake

```diff
@@ -630,6 +692,23 @@ function(cpack_deb_prepare_package_vars)
         endif()
       endif()
     endforeach()
+    set(COMPONENT_DEPENDS "")
+    set(_DEPS "${CPACK_COMPONENT_${_local_component_name}_DEPENDS}")
+    foreach (_PACK ${_DEPS})
+      get_component_package_name(_PACK_NAME "${_PACK}")
+      if(COMPONENT_DEPENDS)
+        set(COMPONENT_DEPENDS "${_PACK_NAME} (= ${CPACK_DEBIAN_PACKAGE_VERSION}), ${COMPONENT_DEPENDS}")
```

Comment:
> I just add our own components first, then all other deps

No, that's further below. Look at the line that this comment is on.

> Do you know any drawbacks of the solution?

Well, I don't think the order of dependencies matters, so it's nothing serious, but it does violate the principle of least astonishment, and for no apparent reason.